src/Build/Definition/ProjectCollection.cs

```diff
@@ -1241,10 +1168,7 @@ public Project LoadProject(XmlReader xmlReader, IDictionary<string, string> glob
         /// </summary>
         public void RegisterLogger(ILogger logger)
         {
-            using (_locker.EnterDisposableWriteLock())
-            {
-                RegisterLoggerInternal(logger);
-            }
+            RegisterLoggerInternal(logger);
```

Comment:
This is just calling `LoggingService.RegisterLogger` which has its own locking. No need to protect it here.